WinCompatibilityPack/WinCompatibilityPack.psm1

```diff
@@ -0,0 +1,793 @@
+
+    Write-Verbose -Verbose:$verboseFlag "Getting remote modules..."
+    # Use Invoke-Command here instead of the -PSSession option on Get-Module because
+    # we're only returning a subset of the data
+    $RemoteModule = @(Invoke-Command -Session $session {
+        (Get-Module -ListAvailable).
+            Where{$_.Name -notin $using:NeverImportList -and $_.Name -like $using:Name} |
+                Select-Object Name, Version })
+
+    Write-Verbose -Verbose:$verboseFlag "Comparing module set..."     
+    Compare-Object $LocalModule $RemoteModule -Property Name,Version |
+        Where-Object SideIndicator -eq "=>"
+}
+
+###########################################################################################
+<#
+.Synopsis
+   Copy modules from the compatibility session that are directly usable in PowerShell Core.
+.DESCRIPTION
+   Copy modules from the compatibility session that are directly usable in PowerShell Core.
+   By default, these modules will be copied to $PSHome/Modules. This can be overridden using
+   the -Destination parameter. Once these modules have been copied, they will be available
+   just like the other native modules for PowerShell Core.
+
+   Note that if there already is a module in the destination corresponding to the module
+   to be copied's name, it will not be copied. 
+.EXAMPLE
+    Copy-WinModule hyper-v -WhatIf -Verbose
+
+    Run the copy command with -WhatIf to see what would be copied to $PSHome/Modules.
+    Also show Verose information.
+.EXAMPLE
+    PS C:\> Copy-WinModule hyper-v -Destination ~/Documents/PowerShell/Modules
+
+    Copy the specified module to your user module directory.
+#>
+function Copy-WinModule
+{
+    [CmdletBinding(SupportsShouldProcess)]
+    [Alias("cpwinm")]
+    [OutputType([void])]
+    Param
+    (
+        # Specifies names or name patterns of modules that will be copied. 
+        # Wildcard characters are permitted.
+        [Parameter(Mandatory=$false,Position=0)]
+        [String[]]
+            $Name="*",
+
+        # If you don't want to use the default compatibility session, use
+        # this parameter to specify the name of the computer on which to create
+        # the compatibility session.
+        # (Defaults to 'localhost')
+        [Parameter()]
+        [String]
+        [Alias("cn")]
+            $ComputerName = $script:DefaultComputerName,
+
+        # Specifies the configuration to connect to when creating the compatibility session
+        # (Defaults to 'Microsoft.PowerShell')
+        [Parameter()]
+        [String]
+            $ConfigurationName = $script:DefaultConfigurationName,
+
+        # If needed, use this parameter to specify credentials for the compatibility session
+        [Parameter()]
+        [PSCredential]
+            $Credential,
+        
+        # The location where compatible modules should be copied to
+        [Parameter()]
+        [String]
+            $Destination = (join-path $PSHOME "Modules")
```

Comment:
I agree.  I think it's only controversial for PSGet since they optimized for DSC early on (where it was required to be in the global path for LCM to find it...) and is not a backwards compatibility problem for them.

Reply:
This is whar @jpsnover had implemented originally but I agree with your concerns so I've changed it to default to the user's module directory and also added some validation logic for the path.